### PR TITLE
[10.0] [BACKPORT] l10n_ch_qr_bill_scan

### PR DESCRIPTION
**Pull request by yvaucher**
This requires:
- [ ] OCA/edi#234

### CODE REVIEW

File: l10n_ch_qr_bill_scan/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
+In order to configure the type of invoice created per partner.
+
+Got to:
```
Comment (gurneyalex) [inline l10n_ch_qr_bill_scan/readme/CONFIGURE.rst:3]:
```suggestion
Go to:
```

File: l10n_ch_qr_bill_scan/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Import Vendor bills with a PayEye scanner.
+
+The QR bills are introduced in Switzerland since the 30th of June 2020.
```
Comment (gurneyalex) [inline l10n_ch_qr_bill_scan/readme/DESCRIPTION.rst:3]:
```suggestion
The QR bills were introduced in Switzerland on the 30th of June 2020.
```

File: l10n_ch_qr_bill_scan/readme/INSTALL.rst [hunk 3]
```diff
@@ -0,0 +1,11 @@
+This module depends on `account_invoice_import` which is located in `edi` and has dependencies on `community-data-files`.
+So the repository `edi` and `community-data-files` must be present on the system.
```
Comment (gurneyalex) [inline l10n_ch_qr_bill_scan/readme/INSTALL.rst:2]:
```suggestion
So the repositories `edi` and `community-data-files` must be present on the system.
```

File: l10n_ch_qr_bill_scan/readme/USAGE.rst [hunk 4]
```diff
@@ -0,0 +1,9 @@
+To import your invoices Launch the wizard `Invoicing` -> `Vendors` -> `Import Vendor Bill`.
```
Comment (gurneyalex) [inline l10n_ch_qr_bill_scan/readme/USAGE.rst:1]:
```suggestion
To import your invoices, launch the wizard `Invoicing` -> `Vendors` -> `Import Vendor Bill`.
```

File: l10n_ch_qr_bill_scan/readme/USAGE.rst [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+In there you can either directly the QR code of the QR-bill or you can import a PDF/image file.
```
Comment (gurneyalex) [inline l10n_ch_qr_bill_scan/readme/USAGE.rst:3]: Missing word, here. "import" ? "scan" ? 
```suggestion
In there you can either directly import the QR code of the QR-bill or you can import a PDF/image file.
```

File: l10n_ch_qr_bill_scan/readme/USAGE.rst [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+
+If it's the first time you import a Bill from a Vendor the wizard will ask you to create the partner and to set a import configuration for that partner
```
Comment (gurneyalex) [inline l10n_ch_qr_bill_scan/readme/USAGE.rst:5]:
```suggestion
If it's the first time you import a Bill from a Vendor the wizard will ask you to create the partner and to set an import configuration for that partner
```

File: l10n_ch_qr_bill_scan/readme/USAGE.rst [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+A configuration let you set which expense account you want the invoice to be imported with.
```
Comment (gurneyalex) [inline l10n_ch_qr_bill_scan/readme/USAGE.rst:6]:
```suggestion
A configuration lets you set which expense account you want the invoice to be imported with.
```

File: l10n_ch_qr_bill_scan/readme/USAGE.rst [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+
+
+To quickly import a serie of invoice, use the button `Import Next Invoice` from the Vendor bill form.
```
Comment (gurneyalex) [inline l10n_ch_qr_bill_scan/readme/USAGE.rst:9]:
```suggestion
To quickly import a series of invoices, use the button `Import Next Invoice` from the Vendor bill form.
```

**Review by gurneyalex — APPROVED** [review]
some improvement in the readme suggested.

File: l10n_ch_qr_bill_scan/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+{
+    "name": "Switzerland - QR-bill scan",
+    "summary": "Scan QR-bill to create vendor bills",
+    "version": "10.0.1.0.0",
+    "author": "Camptocamp,Odoo Community Association (OCA)",
+    "category": "Localization",
+    "website": "https://github.com/OCA/l10n-switzerland",
+    "license": "AGPL-3",
+    "depends": ["l10n_ch", "account_invoice_import"],
```
Comment (i-vyshnevska) [inline l10n_ch_qr_bill_scan/__manifest__.py]: we need python-dependency to have complaining on install
pyzbar==0.1.8
numpy==1.16.2
pdf2image==1.13.1
opencv-python==4.2.0.34

File: l10n_ch_qr_bill_scan/wizard/account_invoice_import.py [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+        # Ignore everything before "SPC"
+        if qr_match.start() > 0:
+            qr_data = qr_data[qr_match.start() :]
+
+        qr_data = qr_data.split("\n")
+
+        try:
+            amount = float(qr_data[QR.AMOUNT])
+        except ValueError:
+            raise UserError(_("The {} "))
```
Comment (i-vyshnevska) [inline l10n_ch_qr_bill_scan/wizard/account_invoice_import.py]:
```suggestion
            raise UserError(_("Amount cannot be properly read from scan data"))
```

**Pull request closed without merging**